server/src/main/java/org/apache/kafka/server/share/metrics/ShareGroupMetrics.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.share.metrics;
+
+import org.apache.kafka.clients.consumer.AcknowledgeType;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.server.metrics.KafkaMetricsGroup;
+
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Meter;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * ShareGroupMetrics is used to track the broker-side metrics for the ShareGroup.
+ */
+public class ShareGroupMetrics implements AutoCloseable {
+    // Rate of records acknowledged per acknowledgement type.
+    private static final String RECORD_ACKNOWLEDGEMENTS_PER_SEC = "RecordAcknowledgementPerSec";
+    // The time in milliseconds to load the share partitions.
+    private static final String PARTITION_LOAD_TIME_MS = "PartitionLoadTimeMs";
+    private static final String ACK_TYPE = "ackType";
```

Comment:
I suggest renaming this because it's specifically a tag. Maybe "ACK_TYPE_TAG".

Reply:
Done.